cranelift/codegen/src/isa/pulley_shared/inst/emit.rs

```diff
@@ -104,8 +104,8 @@ where
         // (with an `EmitIsland`). We check this in debug builds. This is `mut`
         // to allow disabling the check for `JTSequence`, which is always
         // emitted following an `EmitIsland`.
-        let start = sink.cur_offset();
-        pulley_emit(self, sink, emit_info, state, start);
+        let mut start = sink.cur_offset();
```

Comment:
Not having worked on Pulley from the beginning, I'm stepping out on a limb here: it's worrying that once we introduce mutable state during codegen (and later during disassembly) we will have difficulty going back from this. This feels like a better discussion to have with @fitzgen since he may have stronger opinions about this (or not).

Reply:
FWIW this is modeled after the [aarch64 backend](https://github.com/bytecodealliance/wasmtime/blob/2af831a6defed4378a0c6a324d9ecd5fbab58eff/cranelift/codegen/src/isa/aarch64/inst/emit.rs#L720)